pkg/apis/jenkins.io/v1/register.go

```diff
@@ -43,6 +43,7 @@ func addKnownTypes(scheme *runtime.Scheme) error {
 		&BuildPack{},
 		&BuildPackList{},
 		&App{},
+		&Application{},
```

Comment:
I thought so as well, but TBH i am confused by the naming in general.  I picked this up from @agentgonzo who claimed to have discussed creating the new CRD with the `jx` team.

Reply:
I think the `App` CRD was introduced by @pmuir for extensions. It is worth to unify the both CRDs in order to avoid confusion and keep the things simple. We can define a field for `application type` which can hold various values such as `extension`, `microservice`, etc. Then if you need all applications of a certain type, it can be queried by this field.

cc @jstrachan

Reply:
Do you think that a user is not going to be confused by `app` vs `application` being two different things? There are also `plugins`, `addons`, maybe `extensions`.  IMO, the clear way is to adopt `application` everywhere and define different types of applications which clearly describe their purpose.

BTW `jx get applications` has aliases for `app` and `apps`, also there is `jx get plugins` and `jx get addons` just to increase the confusion. 

I think, if we strive to be more consistent and reduce the number of terms, it will simplify also the design. I am a bit concern by the explosion of CRDs.

Reply:
agreed - `jx get app` is gonna be confusing if it means "applications" and not "Apps" (as in addons/extensions to the platform itself).

Maybe we need to move to using a `Project` CRD to represent what we called "Applications" and then use `jx get project` instead so that `jx get app` and `jx get application` returns the equivalent of `jx get addon`

Reply:
so `jx get app == jx get addon` and `jx get project` returns what projects have been added to Jenkins X (via `jx create project / jx import / jx create spring|quickstart|camel` etc

Reply:
we have a general purpose top level `jx create project` now BTW https://jenkins-x.io/commands/jx_create_project/

Reply:
`project` is a good idea. It sounds more generic since a project can host various artifacts such as application or library. The library is not really deployable in Kubernetes, so not an application in runtime. 

What about `plugins`? What's the difference between a plugin and an application/extension? Is it just an extension of `jx` CLI? 

To sum up the current state to the future state:
application/app => project 
addon/extensions => application
plugin => plugin

Reply:
the CloudBees Core UI uses Projects now btw rather than Applications. 

Its tricky as folks may think `App` is an abbreviation of `Applications` so can get confused between totally different concepts (user's microservices/charts they deploy to apps versus Apps)

Reply:
Ah not realised that the UI shows projects now, fair enough.  Just might take a bit of getting used to I guess.